gtsfm/utils/prepare_nerfstudio.py

```diff
@@ -0,0 +1,162 @@
+        out.update(
+            {
+                "k1": float(camera_params[4]),
+                "k2": float(camera_params[5]),
+                "p1": float(camera_params[6]),
+                "p2": float(camera_params[7]),
+            }
+        )
+    if camera_model == CameraModel.OPENCV_FISHEYE:
+        out.update(
+            {
+                "k1": float(camera_params[4]),
+                "k2": float(camera_params[5]),
+                "k3": float(camera_params[6]),
+                "k4": float(camera_params[7]),
+            }
+        )
+
+    out["frames"] = frames
+
+    with open(os.path.join(output_dir, "transforms.json"), "w", encoding="utf-8") as f:
+        json.dump(out, f, indent=4)
+
+    return cameras[1].width, cameras[1].height
+
+
+def resize_and_save_images_for_nerfstudio(images_dir, image_width, image_height, resized_images_dir):
+    """Resizes images to the size used in GTSfM and saves for nerfstudio.
+    Args:
+        images_dir: path to the original images.
+        images_width: int width of resized images
+        images_height: int height of resized images.
+        resized_images_dir: path to directory where the resized images will be saved.
+
+    """
+    image_filenames = os.listdir(images_dir)
+    for image_filename in image_filenames:
+        image_path = os.path.join(images_dir, image_filename)
+        image = io.load_image(image_path)
+        resized_image = images.resize_image(image, image_height, image_width)
+        resized_image_path = os.path.join(resized_images_dir, image_filename)
+        io.save_image(resized_image, resized_image_path)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--results_path", required=True, help="Relative or absolute path to /gtsfm/results/")
+    parser.add_argument(
+        "--camera_model",
+        required=True,
+        help="'perspective' or 'fisheye' corresponding to the OPENCV and OPENCV_FISHEYE camera models at"
+        + " https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/cameras/cameras.py",
+    )
+    parser.add_argument(
+        "--images_dir",
+        required=True,
+        help="Relative or absolute path to the directory of images GTSfM used for reconstruction",
+    )
+    args = parser.parse_args()
+
+    # Create transforms.json, which contains intrinsics and extrinsics
+    results_path = args.results_path
+    cameras_file = os.path.join(results_path, "ba_output", "cameras.txt")
+    images_file = os.path.join(results_path, "ba_output", "images.txt")
+    nerfstudio_input_dir = os.path.join(results_path, "nerfstudio-input")
+    if not os.path.exists(nerfstudio_input_dir):
+        os.makedirs(nerfstudio_input_dir)
+    image_width, image_height = colmap_to_json(
+        cameras_file, images_file, nerfstudio_input_dir, camera_model=CAMERA_MODELS[args.camera_model]
+    )
+
+    # Resize images (as GTSfM does) to match intrinsics in transforms.json
```

Comment:
the size in transforms.json comes from the images.txt and cameras.txt that we generate in gtsfm, so its basically the downsizing that gtsfm does?

Reply:
Yes